lib/web_ui/dev/screenshot_manager.dart

```diff
@@ -0,0 +1,247 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.6
+import 'dart:io' as io;
+import 'dart:convert';
+import 'dart:math';
+
+import 'package:image/image.dart';
+import 'package:path/path.dart' as path;
+import 'package:webkit_inspection_protocol/webkit_inspection_protocol.dart'
+    as wip;
+import 'package:yaml/yaml.dart';
+
+import 'common.dart';
+import 'environment.dart';
+import 'safari_installation.dart';
+import 'utils.dart';
+
+/// [ScreenshotManager] implementation for Chrome.
+///
+/// This manager can be used for both macOS and Linux.
+// TODO: https://github.com/flutter/flutter/issues/65673
+class ChromeScreenshotManager extends ScreenshotManager {
+  String get filenameSuffix => '';
+
+  /// Capture a screenshot of the web content.
+  ///
+  /// Uses Webkit Inspection Protocol server's `captureScreenshot` API.
+  ///
+  /// [region] is used to decide which part of the web content will be used in
+  /// test image. It includes starting coordinate x,y as well as height and
+  /// width of the area to capture.
+  Future<Image> capture(Rectangle region) async {
```

Comment:
`Rectangle<double>`?

Reply:
Is it double though? As far as I see it's only int. What advantage it ads to make the capture's rectangle double?

The crop method only accepts int so making the one on _scaleScreenshotRegion will also result in type conversions.